src/scripts/Add-Scope.ps1

```diff
@@ -0,0 +1,241 @@
+    $payload = $payload.Replace("{4}", $ApiParams.TimePeriodFrom)
+    $payload = $payload.Replace("{5}", $ApiParams.TimePeriodTo)
+    $payload = $payload.Replace("{6}", $ApiParams.PartitionData)
+  }
+  else {
+    $payload = '{
+      "properties": {
+        "schedule": {
+          "status": "{8}",
+          "recurrence": "{7}",
+          "recurrencePeriod": {
+            "from": "{6}",
+            "to": "{9}"
+          }
+        },
+        "partitionData": "{5}",
+        "format": "Csv",
+        "deliveryInfo": {
+          "destination": {
+            "resourceId": "{0}",
+            "container": "{1}",
+            "rootFolderPath": "{2}"
+          }
+        },
+        "definition": {
+          "type": "{3}",
+          "timeframe": "{4}",
+          "dataSet": {
+            "granularity": "Daily"
+          }
+        }
+      }
+    }'
+
+    $payload = $payload.Replace("{0}", $ApiParams.DestinationResourceId)
+    $payload = $payload.Replace("{1}", $ApiParams.DestinationContainer)
+    $payload = $payload.Replace("{2}", $ApiParams.DestinationRootFolderPath)
+    $payload = $payload.Replace("{3}", $ApiParams.DefinitionType)
+    $payload = $payload.Replace("{4}", $ApiParams.DefinitionTimeframe)
+    $payload = $payload.Replace("{5}", $ApiParams.PartitionData)
+    $payload = $payload.Replace("{6}", $ApiParams.RecurrencePeriodFrom)
+    $payload = $payload.Replace("{9}", $ApiParams.RecurrencePeriodTo)
+    $payload = $payload.Replace("{7}", $ApiParams.ScheduleRecurrence)
+    $payload = $payload.Replace("{8}", $ApiParams.ScheduleStatus)
+  }
+  
+  $apiResult = Invoke-AzRestMethod -Uri $uri -Method PUT -Payload $payload
+  if ($apiResult.StatusCode -ne "201") {
+    $apiResult
+    Throw "Cost Management API call failed"
+  }
+}
+
+function Set-CostManagementExport {
+  param (
+    $ExportParams,
+    [bool]$Start = $false
+  )
+
+  Write-DebugInfo $ExportParams
+  Set-CostManagementApi -ApiParams $ExportParams
+  if ($Start) {
+    Start-Sleep -Seconds $SleepInterval
+    Invoke-AzCostManagementExecuteExport -ExportName $ExportParams.Name -Scope $ExportParams.Scope
+    Start-Sleep -Seconds $SleepInterval
+    $currentStatus = $null
+    $currentStatus = (Get-AzCostManagementExport -Name $ExportParams.Name -Scope $ExportParams.Scope -Expand runHistory).RunHistory.Value[0].Status
+    [int]$loop = 0
+    while ($currentStatus -eq "InProgress") {
+      if ($loop -ge $TimeOutMinutes) {
+        $currentStatus = "TimedOut"
+      }
+      else {
+        Start-Sleep -Seconds 60
+        $loop++
+        $currentStatus = (Get-AzCostManagementExport -Name $ExportParams.Name -Scope $ExportParams.Scope -Expand runHistory).RunHistory.Value[0].Status
+        Write-Host ("{0}    {1}    {2}" -f (get-date), $ExportParams.Name, $currentStatus)
+      }
+    }
+
+    Write-Host ("{0}    {1}    {2}" -f (get-date), $ExportParams.Name, $currentStatus)
+  }
+}
+
+if ($Future) {
```

Comment:
Add a comment to explain what this is for. Link to an issue, if we have one to track the work.

Reply:
This is so that I can just pull historical information without needing to reconfigure the recurring future exports.